components/canvas/webgl_mode/inprocess.rs

```diff
@@ -94,13 +107,34 @@ impl WebGLExternalImageApi for WebGLExternalImages {
         self.webgl_channel
             .send(WebGLMsg::Lock(ctx_id, self.lock_channel.0.clone()))
             .unwrap();
-        let (image_id, size, gl_sync) = self.lock_channel.1.recv().unwrap();
-        // The next glWaitSync call is run on the WR thread and it's used to synchronize the two
-        // flows of OpenGL commands in order to avoid WR using a semi-ready WebGL texture.
-        // glWaitSync doesn't block WR thread, it affects only internal OpenGL subsystem.
-        self.webrender_gl
-            .wait_sync(gl_sync as gl::GLsync, 0, gl::TIMEOUT_IGNORED);
-        (image_id, size)
+        let (image_id, size, io_surface_id, gl_sync) = self.lock_channel.1.recv().unwrap();
+
+        let texture_id = match io_surface_id {
+            Some(io_surface_id) => {
+                let texture_id = match self.textures.entry(io_surface_id) {
+                    Entry::Occupied(o) => *o.get(),
```

Comment:
Nit: can we use `Entry::or_insert_with()` here?

Reply:
As I remember we can't because of the borrow of `self` [here](https://github.com/servo/servo/pull/23509/files/055fafb452aa053d520215949100d3a62a436aaa#diff-250070c6c5a38c7f9fa0f5b3c101f68bR118), but I will check it again.